src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService_Helpers.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.InheritanceMargin
+{
+    internal partial class AbstractInheritanceMarginService
+    {
+        private static async Task<InheritanceMemberItem> CreateInheritanceMemberInfoAsync(
+            Solution solution,
+            INamedTypeSymbol memberSymbol,
+            int lineNumber,
+            ImmutableArray<ISymbol> baseSymbols,
+            ImmutableArray<ISymbol> derivedTypesSymbols,
+            CancellationToken cancellationToken)
+        {
+            var baseSymbolItems = await baseSymbols
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Implementing, cancellationToken))
+                .ConfigureAwait(false);
+
+            var derivedTypeItems = await derivedTypesSymbols
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Implemented, cancellationToken))
+                .ConfigureAwait(false);
+
+            return new InheritanceMemberItem(
+                lineNumber,
+                memberSymbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                memberSymbol.GetGlyph(),
+                baseSymbolItems.Concat(derivedTypeItems));
+        }
+
+        private static async ValueTask<InheritanceTargetItem> CreateInheritanceItemAsync(
+            Solution solution,
+            ISymbol targetSymbol,
+            InheritanceRelationship inheritanceRelationshipWithOriginalMember,
+            CancellationToken cancellationToken)
+        {
+            var definition = await targetSymbol.ToClassifiedDefinitionItemAsync(
+                solution,
+                isPrimary: true,
+                includeHiddenLocations: false,
+                FindReferencesSearchOptions.Default,
+                cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
ooh, do we actually show these as classified in the dropdown?  that would be nice.

Reply:
That's what I am expecting originally, but it might not be true for the its first version (I look the margin pops up and no one is providing the coloring). I will mark here and change it to unclassified one.